galaxy_importer/ansible_test/builders/pulp_build.py

```diff
@@ -15,7 +15,271 @@
+        except AttributeError:
+            spath = os.path.join(
+                os.path.dirname(constants.ROOT_DIR.rstrip('galaxy_importer')),
+                f'{self.metadata.namespace}-{self.metadata.name}-{self.metadata.version}.tar.gz')
+            path = os.path.join(
+                os.path.dirname(os.path.realpath(__file__)),
+                'archive.tar.gz')
+            copy(spath, path)
+        return path
+
+    def setup_dockerfile(self):
+        # Make a copy of Dockerfile
+        copyfile(
+            self.shared_dockerfile,
+            self.dockerfile)
+
+        # Update Dockerfile to download and copy Collection into the image.
+        with open(self.dockerfile, 'a') as f:
+            f.write("\nCOPY archive.tar.gz /archive/archive.tar.gz\n")
+
+    def build(self):
+        self.container_repository_href = self._get_container_repository_href()
+        self.dockerfile_artifact_href = self._get_artifact_href(self.dockerfile)
+        self.collection_artifact_href = self._get_artifact_href(self.collection_file)
+        self.container_distribution_href = self._get_container_distribution_href()
+        self._build_image()
+
+    def get_registry_href(self):
+        return self._get_registry_href()
+
+    def _build_image(self):
+        artifacts = json.dumps({f'{self.collection_artifact_href}': 'archive.tar.gz'})
+        r = requests.post(
+            url=f'{self.api_url}{self.container_repository_href}build_image/',
+            auth=self.basic_auth,
+            data={
+                'containerfile_artifact': self.dockerfile_artifact_href,
+                # 'containerfile': open(self.dockerfile, 'rb'),
+                'tag': 'latest',
+                'artifacts': artifacts
+            }
+            # files={'containerfile': open(self.dockerfile, 'rb')}
+        )
+        if r.status_code != 202:
+            self.cleanup()
+            raise exceptions.AnsibleTestError(
+                f'Could not build Image: {r.status_code} {r.reason} {r.text}')
+        return self._wait_for_image_build(r.json()['task'])
+
+    def _wait_for_image_build(self, task_href):
+        for i in range(API_CHECK_RETRIES):
+            r = requests.get(
+                url=f'{self.api_url}{task_href}',
+                auth=self.basic_auth)
+            self.log.info('Building image...')
+            status = r.json()['state']
+            if status == 'failed' or status == 'canceled':
+                self.cleanup()
+                raise exceptions.AnsibleTestError(
+                    f'Could not create image: \
+                        {r.status_code} {r.reason} {r.text}')
+            elif status == 'completed':
+                self.log.info('Image successfully built')
+                return
+            time.sleep(API_CHECK_DELAY_SECONDS)
+        if i >= API_CHECK_DELAY_SECONDS:
+            self.cleanup()
+            raise exceptions.AnsibleTestError('Timed out building image')
```

Comment:
```suggestion
        self.cleanup()
        raise exceptions.AnsibleTestError('Timed out building image')
```

Reply:
i don't think the comparison of `API_CHECK_RETRIES` to `API_CHECK_DELAY_SECONDS` yields any useful information and is always true when `API_CHECK_DELAY_SECONDS` equals 1